src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/cpp/CppCompletionManager.java

```diff
@@ -210,21 +210,27 @@ else if (event.getKeyCode() == 113 // F2
             case KeyCodes.KEY_N: return popup.selectNext();
             }
          }
-             
+         
+         // allow navigation with arrow keys
+         // (needs to be handled here as the popup will not be explicitly focused)
+         if (modifier == KeyboardShortcut.NONE)
+         {
+            switch (keyCode)
+            {
+            case KeyCodes.KEY_UP:    return popup.selectPrev();
+            case KeyCodes.KEY_DOWN:  return popup.selectNext();
+            case KeyCodes.KEY_ENTER: return popup.acceptSelected();
+            case KeyCodes.KEY_TAB:   return popup.acceptSelected();
+            }
+         }
+         
```

Comment:
For parity with the old behavior, please also add support for home, end, PgUp and PgDn. This is what got removed, essentially:

```
                     case KeyCodes.KEY_PAGEDOWN:
                        moveSelectionFwd(5);
                        break;
                     case KeyCodes.KEY_PAGEUP:
                        moveSelectionBwd(5);
                        break;
                     case KeyCodes.KEY_HOME:
                        selectFirst();
                        break;
                     case KeyCodes.KEY_END:
                        selectLast();
                        break;